### PR TITLE
Polls controllers test

### PR DESCRIPTION
**Pull request by aalthaf**
- Added delete all methods to reset database
- Added test cases for polls and few others
- Added some response status codes

Additional notes: 

-  cross-env dependency to set environment variables properly

### CODE REVIEW

File: server/db/reset.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  const delUser = await userControllers.deleteAll();
+  const delFriends = await friendsControllers.deleteAll();
+  const delFriendsLists = await friendsListControllers.deleteAll();
+  const delPolls = await pollControllers.deleteAll();
+  const delVotes = await voteControllers.deleteAll();
+
+  return {
+    delUser,
+    delFriends,
+    delFriends,
```
Comment (yoniakabecky) [inline server/db/reset.js]: Is there a reason why calling `delFriends` twice??

File: server/test/app.test.js [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+            expect(err).to.be.null;
+            expect(res).to.have.status(201);
+            res.body.should.have.property("user");
+            res.body.user.should.have.property("email").eql(userFriend.email);
+            res.body.should.have.property("token");
+            res.body.user.should.have.property("_id");
+            friendId = res.body.user._id;
+            friendToken = res.body.token;
+            done();
+          });
```
Comment (yoniakabecky) [inline server/test/app.test.js]: Just wondering🤔, can you reduce repeating code??

**Review by yoniakabecky — APPROVED** [review]
Nicely done 👍

File: server/test/app.test.js [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+// 9) Edit poll (update poll)
+// 10) Delete poll
+
+const expectToBe = (body, actual) => {
+  expect(body).to.satisfy((arr) =>
+    arr.some((poll) => expect(poll).to.deep.include(actual))
+  );
+};
+
+reset()
```
Comment (yoniakabecky) [inline server/test/app.test.js:37]: Do we need to reset all DB every test run?

File: server/db/reset.js (hunk 1 above)
Comment (aalthaf) [reply server/db/reset.js]: Nope my mistake .

File: server/test/app.test.js (hunk 3 above)
Comment (aalthaf) [reply server/test/app.test.js:37]: User sign up can fail due to the possibility of user being already present from previous test runs or even manually. So I'm resetting all data.

**Pull request merged**